ape_starknet/provider.py

```diff
@@ -268,17 +268,18 @@ def send_transaction(self, txn: TransactionAPI, token: Optional[str] = None) ->
             receipt.returndata = returndata.copy()
             abi = txn.method_abi
 
-            if txn.original_method_abi:
-                # When that special attribute is set means the transation came from an
-                # account-specific call: it implies the original method ABI was replaced
-                # with the specific execute ABI in BaseStarknetAccount.prepare_transaction(),
+            if txn.original_method_abi and txn.original_full_abi:
+                # When those special attributes are set means the transaction came from an
+                # account-specific call: it implies original ABIs were replaced
+                # with account-specific ABIs in BaseStarknetAccount.prepare_transaction(),
                 # and that the return data is always prefixed with the number of items.
                 # We need to restore the former, and remove the later.
                 abi = txn.original_method_abi
                 returndata = returndata[1:]
-
-            return_value = self.starknet.decode_returndata(abi, returndata)
-            receipt.return_value = return_value
+                return_value = self.starknet.decode_returndata(
+                    abi, returndata, txn.original_full_abi
+                )
+                receipt.return_value = return_value
```

Comment:
This is over indented :thinking: ?
```suggestion
            return_value = self.starknet.decode_returndata(
                abi, returndata, full_abi=txn.original_full_abi
            )
            receipt.return_value = return_value
```